### PR TITLE
Fix for the issue when machine cannot be started second time, and better nspawn logging

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
travis fails, but I don't see any error. Same with centos ci.

**Comment by keszybz** [comment]
@poettering, @yuwata — please review this if you have a moment. I think the nspawn issue needs to be fixed before -rc1.

File: src/resolve/resolved-dnstls-gnutls.c [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #include "resolved-dns-stream.h"
 #include "resolved-dnstls.h"
 
-#define PRIORTY_STRING "NORMAL:-VERS-ALL:+VERS-TLS1.3:+VERS-TLS1.2"
+#define TLS_PROTOCOL_PRIORITY "NORMAL:-VERS-ALL:+VERS-TLS1.3:+VERS-TLS1.2"
```
Comment (poettering) [inline src/resolve/resolved-dnstls-gnutls.c:12]: ouch!

File: src/nspawn/nspawn.c [hunk 2]
```diff
@@ -492,6 +492,44 @@ static int detect_unified_cgroup_hierarchy_from_image(const char *directory) {
...
+                if (streq(t, "all"))
+                        *mask = (uint64_t) -1;
+                else {
+                        r = capability_from_name(t);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse capability %s.", t);
+
+                        *mask |= 1ULL << r;
+                }
+        }
```
Comment (poettering) [inline src/nspawn/nspawn.c:529]: hmm a function that clobbers the return value even on failure? please don't!

File: src/nspawn/nspawn-seccomp.c [hunk 3]
```diff
@@ -153,7 +152,7 @@ static int seccomp_add_default_syscall_filter(
         }
 
         STRV_FOREACH(p, syscall_whitelist) {
-                r = seccomp_add_syscall_filter_item(ctx, *p, SCMP_ACT_ALLOW, syscall_blacklist, false);
+                r = seccomp_add_syscall_filter_item(ctx, *p, SCMP_ACT_ALLOW, syscall_blacklist, true);
```
Comment (poettering) [inline src/nspawn/nspawn-seccomp.c:155]: (hmm, what was the rationale for making the logging optional in the first place? I mean debug logging should always be OK?)

Comment (keszybz) [reply src/nspawn/nspawn-seccomp.c:155]: ~~True, I shouldn't have done that.~~

I think this logging was resulting in excessive lines when the default filters were initialized. But in this code path it should be OK.

File: src/machine/machine.c [hunk 4]
```diff
@@ -357,28 +361,25 @@ static int machine_start_scope(
...
+        r = sd_bus_message_append(m, "(sv)", "Slice", "s", SPECIAL_MACHINE_SLICE);
+        if (r < 0)
+                return r;
 
-        if (!isempty(description)) {
-                r = sd_bus_message_append(m, "(sv)", "Description", "s", description);
-                if (r < 0)
-                        return r;
-        }
+        const char *description = strjoina(machine->class == MACHINE_VM ? "Virtual Machine " : "Container ", machine->name);
```
Comment (poettering) [inline src/machine/machine.c]: uhhh i don't know what this line really means: you do two stack allocations here, one via strjoina() and one via the variable declaration. Is this even defined in its behaviour? I'd really split that into two lines...

Comment (keszybz) [reply src/machine/machine.c]: It's like saying `const char *foo, *bar`: the compiler also does two allocations on a single line. I don't think we need to care about the details of what exactly happens there. I'm sure there are some rules that specify in which order the allocations should happen, but then the compiler is probably free to reorder them when optimizing anyway. I'm pretty sure the pointer is optimized away.

I'll make this into two lines, but I don't this matters at all. If anything, separating the declaration and initialization increases (very slightly) the chance of an error.

File: src/nspawn/nspawn.c (hunk 2 above)
Comment (keszybz) [reply src/nspawn/nspawn.c:529]: True, I shouldn't have done that.

**Comment by keszybz** [comment]
Updated.

File: src/machine/machine.c (hunk 4 above)
Comment (poettering) [reply src/machine/machine.c]: still smells fishy, but ok

**Comment by keszybz** [comment]
bionic-arm64 is being slow. I don't think we need to wait for it. Everything else passes.

**Pull request merged**